### PR TITLE
Allow global cache polling interval to be configurable (See #427)

### PR DESCRIPTION
**Pull request by prologic**

### CODE REVIEW

**Review by der-lyse — CHANGES_REQUESTED** [review]
Hmm, this kinda works, but there are several things which are not as I expect them to work:

1. The _internal/jobs.go:30_ defines, that the `UpdateFeedsJob` is executed every five minutes. So when starting twtd with a fetch interval less than five minutes, e.g. `--fetch-interval 10s`, the setting has no effect. You can't fetch more often than once every five minutes. It's probably okay, but at least it should be documented somewhere. More on that later.
2. I patched the _internal/jobs.go:30_, so that it will fetch every ten seconds, to make things more obvious and also added info log messages when the job is skipped and when it is executed.
    1. Choosing `--fetch-interval 20s` executes the startup job and right after that also the first cron job, without waiting at least 20s as witnessed by the timestamps. Stripped the log to the important parts for brevity:
        ```
        $ make dev "FLAGS=--fetch-interval 20s"
        Running in debug mode...
        twt version 0.1.0@cb60653
        ./twtd -D -O -R --fetch-interval 20s
        …
        INFO[0001] Started background job UpdateFeeds (@every 10s)
        …
        INFO[0001] Fetch Interval: 20s
        …
        INFO[0006] running startup jobs
        INFO[0006] running UpdateFeeds now...
        INFO[0006] !!! running feed update 2021-06-11T19:29:57+02:00 !!!
        INFO[0006] updating feeds for 1 users and  2 feeds
        INFO[0006] updating 7 sources
        …
        INFO[0011] !!! running feed update 2021-06-11T19:30:02+02:00 !!!
        INFO[0011] updating feeds for 1 users and  2 feeds
        INFO[0011] updating 7 sources
        …
        ```
        Between 19:29:57 and 19:30:02 are just five seconds, which is way less than the configured 20 seconds.
    2. Also in fact the feeds are fetched only every 30 seconds rather than every 20 seconds:
        ```
        …
        INFO[0021] ### skipping feed update 2021-06-11T19:30:12+02:00 ###
        INFO[0031] ### skipping feed update 2021-06-11T19:30:22+02:00 ###
        INFO[0041] !!! running  feed update 2021-06-11T19:30:32+02:00 !!!
        INFO[0041] updating feeds for 1 users and  2 feeds
        INFO[0041] updating 7 sources
        …
        INFO[0051] ### skipping feed update 2021-06-11T19:30:42+02:00 ###
        INFO[0061] ### skipping feed update 2021-06-11T19:30:52+02:00 ###
        INFO[0071] !!!  running feed update 2021-06-11T19:31:02+02:00 !!!
        INFO[0071] updating feeds for 1 users and  2 feeds
        INFO[0071] updating 7 sources
        …
        ```
    3. Every other fetch cycle is skipped as well, when matching the `--fetch-interval 10s` with the job's `@every 10s`:
        ```
        make dev "FLAGS=--fetch-interval 10s"
        Running in debug mode...
        twt version 0.1.0@cb60653
        ./twtd -D -O -R --fetch-interval 10s
        …
        INFO[0001] Started background job UpdateFeeds (@every 10s)
        …
        INFO[0001] Fetch Interval: 10s
        …
        INFO[0006] running UpdateFeeds now...
        INFO[0006] !!! running feed update 2021-06-11T19:36:23+02:00 !!!
        INFO[0006] updating feeds for 1 users and  2 feeds
        INFO[0006] updating 7 sources
        …
        INFO[0011] !!! running feed update 2021-06-11T19:36:28+02:00 !!!
        INFO[0011] updating feeds for 1 users and  2 feeds
        INFO[0011] updating 7 sources
        …
        INFO[0021] ### skipping feed update 2021-06-11T19:36:38+02:00 ###
        INFO[0031] !!! running  feed update 2021-06-11T19:36:48+02:00 !!!
        INFO[0031] updating feeds for 1 users and  2 feeds
        INFO[0031] updating 7 sources
        …
        INFO[0041] ### skipping feed update 2021-06-11T19:36:58+02:00 ###
        INFO[0051] !!! running  feed update 2021-06-11T19:37:08+02:00 !!!
        INFO[0051] updating feeds for 1 users and  2 feeds
        INFO[0051] updating 7 sources
        …
        ```

Maybe this new `--fetch-interval` setting should be just used in the job's schedule without any extra skipping logic? I didn't try it out, though. This was just an idea I had.

The problem that the startup job's last execution time is not correctly transferred to the periodically scheduled job is a different thing.

**Comment by prologic** [comment]
Thanks @der-lyse 🙇‍♂️ changing the job's actual schedule was more work but I agree it is the better approach 

I'll make the change today

**Comment by prologic** [comment]
@der-lyse Okay this should do the trick nicely.

File: internal/jobs.go [hunk 1]
```diff
@@ -23,10 +23,10 @@ var (
 	StartupJobs map[string]JobSpec
 )
 
-func init() {
+func InitJobs(conf *Config) {
```
Comment (der-lyse) [inline internal/jobs.go:26]: This function is never called, breaking everything, so no job is ever run.

**Review by der-lyse — CHANGES_REQUESTED** [review]
The new approach looks way better I believe. However, twtd doesn't really work at the moment due to a bug.

File: internal/jobs.go (hunk 1 above)
Comment (der-lyse) [reply internal/jobs.go:26]: I fixed it and also referenced the

**Review by der-lyse — APPROVED** [review]
Please have a look at my fix before merging.

**Comment by prologic** [comment]
@der-lyse Haha I forgot to actually call `InitJob`()` d'oh :D Your change/fix looks good Merging!

**Pull request merged**